expertsystem/data.py

```diff
@@ -75,36 +78,115 @@ def magnitude(self) -> float:
         return self.__magnitude
 
     @property
-    def projection(self) -> float:
+    def projection(self) -> Optional[float]:
         return self.__projection
 
     def __hash__(self) -> int:
         return hash(repr(self))
 
 
-class Particle(NamedTuple):
-    """Immutable data container for particle info."""
+class QuantumState(NamedTuple):
+    """Collection of properties defining a quantum mechanical state.
+
+    Related to `.Particle`, but can contain more specific quantum state
+    information, such as `.Spin` projection.
+    """
 
-    name: str
-    pid: int
     charge: int
-    spin: float
-    mass: float
+    spin: Spin
+    isospin: Optional[Spin] = None
     strangeness: int = 0
     charmness: int = 0
     bottomness: int = 0
     topness: int = 0
     baryon_number: int = 0
-    electron_number: int = 0
-    muon_number: int = 0
-    tau_number: int = 0
-    width: Optional[float] = None
-    isospin: Optional[Spin] = None
+    electron_lepton_number: int = 0
+    muon_lepton_number: int = 0
+    tau_lepton_number: int = 0
     parity: Optional[Parity] = None
     c_parity: Optional[Parity] = None
     g_parity: Optional[Parity] = None
 
 
+class ComplexEnergyState:
+    """Pole in the complex energy plane, with quantum numbers."""
+
+    def __init__(self, energy: complex, state: QuantumState):
+        self.__energy = complex(energy)
+        self.state: QuantumState = state
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, ComplexEnergyState):
+            return self == other
+        if isinstance(other, QuantumState):
+            return self.state == other
```

Comment:
I think that might just cause troubles. So if only the state should be compared then `.state` should be used in the call

Reply:
bc9976498ec4450626e46f6a4cd7796d6e074246